core-blocks/footnotes/editor.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import { get, isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { withSelect } from '@wordpress/data';
+import { RichText } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+class FootnotesEditor extends Component {
+	constructor( props ) {
+		super( ...arguments );
+
+		this.setFootnotesInOrder( props.footnotesOrder );
+		this.state = {
+			editable: null,
+		};
+	}
+
+	setFootnotesInOrder( footnotesOrder ) {
+		const { attributes, setAttributes } = this.props;
+
+		const footnotes = footnotesOrder.map( ( { id } ) => {
+			return this.getFootnoteById( attributes.footnotes, id );
+		} );
+
+		setAttributes( { footnotes } );
+	}
+
+	getFootnoteById( footnotes, footnoteUid ) {
+		const filteredFootnotes = footnotes.filter(
+			( footnote ) => footnote.id === footnoteUid );
+
+		return get( filteredFootnotes, [ 0 ], { id: footnoteUid, text: '' } );
+	}
+
+	onChange( footnoteUid ) {
+		return ( nextValue ) => {
+			const { attributes, footnotesOrder, setAttributes } = this.props;
+
+			const nextFootnotes = footnotesOrder.map( ( { id } ) => {
+				if ( id === footnoteUid ) {
+					return {
+						id,
+						text: nextValue,
+					};
+				}
+
+				return this.getFootnoteById( attributes.footnotes, id );
+			} );
+
+			setAttributes( {
+				footnotes: nextFootnotes,
+			} );
+		};
+	}
+
+	onSetActiveEditable( id ) {
+		return () => {
+			this.setState( { editable: id } );
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		const { footnotesOrder } = this.props;
+		const nextFootnotesOrder = nextProps.footnotesOrder;
+
+		if ( ! isEqual( footnotesOrder, nextFootnotesOrder ) ) {
```

Comment:
Do you think we can avoid the deep comparison here?

Reply:
I can't think of any way to avoid this comparison for now.

What we could easily do is to store these two arrays as an array of strings instead of an array of objects (from `[ { id: 'abcd' }, ... ]` to `[ 'abcd', ... ]`). That will make the comparison a little bit faster.

Reply:
I refactored the footnotes block so this comparison is no longer necessary (11794cc28056a76e5e116bf0ed0e6ff0b77cfd60).